crates/astria-sequencer-relayer/src/relayer/read.rs

```diff
@@ -233,11 +233,12 @@ async fn fetch_block(
     .await
     .wrap_err("retry attempts exhausted; bailing")?;
 
-    state.set_sequencer_connected(true);
-
-    let block = SequencerBlock::try_from_raw(block.into_inner())
-        .wrap_err("failed to parse raw proto block from grpc response")?;
+    let maybe_block = SequencerBlock::try_from_raw(block.into_inner())
+        .wrap_err("failed to parse raw proto block from grpc response");
+    state.set_sequencer_connected(maybe_block.is_ok());
```

Comment:
This is the only change to production code, so worth highlighting :)

I think this is the correct behaviour, since if we return an error here, we exit the `select!` loop in `Relayer::run` and are hung until externally killed.  While in that state, I think it makes sense to have the sequencer noted as disconnected so that the `/healthz` endpoint reports degraded health.

Reply:
Are we hanging until externally killed or do we just shut down gracefully?

Reply:
Hanging [here](https://github.com/astriaorg/astria/blob/73852b9476889a989ab45258003d1270cc42d90a/crates/astria-sequencer-relayer/src/relayer/mod.rs#L225).  We could do `self.shutdown_token.cancel();` just before that if we do want to actually shut down?

Reply:
I see - yeah. Good catch. We should do that. It's outside the scope of this PR I'd say, but `relayer` knows that if the read task fails no progress will be made. So it should just shut everything down.